src/lib/components/ui/input/date/picker.svelte

```diff
@@ -22,12 +24,18 @@
 		helperText,
 		value = $bindable(),
 		style,
+		onchange,
 	}: Props = $props()
 
 	let showDatePicker = $state(false)
 
 	const labelVariant = $derived(dimension === 'small' ? 'small' : 'default')
-	const dateString = $derived(value ? value.toISOString().slice(0, 10) : '')
+	const dateString = $derived(value ? formatDate(value) : '')
+
+	$effect(() => {
+		value //eslint-disable-line @typescript-eslint/no-unused-expressions
+		if (onchange) untrack(() => onchange())
```

Comment:
Maybe this should be a bit more precise because I am unsure if `onchange` will always be untracked. See https://github.com/sveltejs/svelte/issues/9451#issuecomment-1811369546

```suggestion
		untrack(() => { if (onchange) onchange() })
```